deisctl/backend/fleet/list_machines.go

```diff
@@ -0,0 +1,76 @@
+package fleet
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/coreos/fleet/machine"
+)
+
+const (
+	defaultListMachinesFields = "machine,ip,metadata"
+)
+
+var (
+	listMachinesFields = map[string]machineToField{
+		"machine": func(ms *machine.MachineState, full bool) string {
+			return machineIDLegend(*ms, full)
+		},
+		"ip": func(ms *machine.MachineState, full bool) string {
+			if len(ms.PublicIP) == 0 {
+				return "-"
+			}
+			return ms.PublicIP
+		},
+		"metadata": func(ms *machine.MachineState, full bool) string {
+			if len(ms.Metadata) == 0 {
+				return "-"
+			}
+			return formatMetadata(ms.Metadata)
+		},
+	}
+)
```

Comment:
This is clever using a map of functions to handle each field.  Looking at the rest of the code and docstrings tho, I don't see any supported options for limiting what fields are returned or requesting additional fields.  Are there plans to do so?  If not, I'm almost wondering if there's some complexity that we could pare down here by just moving this logic right into `printMachines()`